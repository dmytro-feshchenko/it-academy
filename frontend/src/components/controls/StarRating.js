import React from 'react';
import { Rating } from 'semantic-ui-react';

/**
 * Star rating
 * Defines rating bar with stars
 * The stars are selectable.
 */
export default class StarRating extends React.Component {
    render() {
        return (
            <div className="star-rating">
                <Rating maxRating="5" />
            </div>
        )
    }
}