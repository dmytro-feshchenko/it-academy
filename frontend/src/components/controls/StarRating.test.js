import React from 'react';
import renderer from 'react-test-renderer';
import StarRating from './StarRating';

it("renders correctly", () => {

    const tree = renderer.create(
        <StarRating />
    );
    expect(tree).toMatchSnapshot();
});