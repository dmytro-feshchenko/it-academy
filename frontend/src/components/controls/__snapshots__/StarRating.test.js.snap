// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="star-rating"
>
  <div
    className="ui rating"
    onMouseLeave={[Function]}
    role="radiogroup"
  >
    <div
      aria-checked={false}
      aria-posinset={1}
      aria-setsize="5"
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <div
      aria-checked={false}
      aria-posinset={2}
      aria-setsize="5"
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <div
      aria-checked={false}
      aria-posinset={3}
      aria-setsize="5"
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <div
      aria-checked={false}
      aria-posinset={4}
      aria-setsize="5"
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <div
      aria-checked={false}
      aria-posinset={5}
      aria-setsize="5"
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
  </div>
</div>
`;
